server/routes/v1/user.ts

```diff
@@ -91,44 +91,48 @@ export async function signUpHandler(req: Request, res: Response) {
   res.status(response.status).json(response.data);
 }
 
+async function sendPassResetHandler(req: Request, res: Response) {
+  const result = await uContr.sendResetEmail(req.body.email);
+  if (result.data) {
+    console.error(result.data.message);
+    res.status(result.status).json(result.data);
+    return;
+  }
+  res.status(result.status).json();
+}
+
 async function confirmEmailHandler(req: Request, res: Response) {
-  const token = req.query.c as string;
+  const token = req.body.token as string;
 
   if (!token) {
-    res.status(400).json({ code: 400, message: 'Missing token.' });
+    res.status(400).json({ message: 'Missing token.' });
     return;
   }
 
   if (await uContr.confirmEmail(token)) {
     res.status(204).json();
   } else {
-    res
-      .status(400)
-      .json({ code: 400, message: 'Token is invalid or expired.' });
+    res.status(400).json({ message: 'Token is invalid or expired.' });
   }
 }
 
 async function resetPassHandler(req: Request, res: Response) {
-  if (!req.query.r || !req.body.password || !req.body.password_confirmation) {
+  if (!req.body.token || !req.body.password) {
```

Comment:
Password reset and account confirmation should both take a password and password confirmation.

Reply:
account confirmation shouldn't require a password since all it does is confirming the email of an account with a given token

Reply:
This is the handler for resetting a password not account confirmation.

If by account confirmation you mean email verification, I don't think that should take a password and password confirmation, since all you really do is click a "Confirm email address" button.